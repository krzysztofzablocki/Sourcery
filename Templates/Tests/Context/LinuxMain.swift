//
//  LinuxMain.swift
//  Templates
//
//  Created by Anton Domashnev on 17.05.17.
//  Copyright Â© 2017 Pixle. All rights reserved.
//

import Foundation
import XCTest

class AutoInjectionTests: XCTestCase {
    func testThatItResolvesAutoInjectedDependencies() {
        XCTAssertTrue(true)
    }

    func testThatItDoesntResolveAutoInjectedDependencies() {
        XCTAssertTrue(true)
    }
}

class AutoWiringTests: XCTestCase {
    func testThatItCanResolveWithAutoWiring() {
        XCTAssertTrue(true)
    }

    func testThatItCanNotResolveWithAutoWiring() {
        XCTAssertTrue(true)
    }
}

// sourcery: disableTests
class DisabledTests: XCTestCase {
    func testThatItResolvesDisabledTestsAnnotation() {
        XCTAssertTrue(true)
    }
}
