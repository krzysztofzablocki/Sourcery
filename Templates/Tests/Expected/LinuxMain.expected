import XCTest

extension AutoInjectionTests {
  static var allTests: [(String, (AutoInjectionTests) -> () throws -> Void)] = [
    ("testThatItResolvesAutoInjectedDependencies", testThatItResolvesAutoInjectedDependencies),
    ("testThatItDoesntResolveAutoInjectedDependencies", testThatItDoesntResolveAutoInjectedDependencies)
  ]
}
extension AutoWiringTests {
  static var allTests: [(String, (AutoWiringTests) -> () throws -> Void)] = [
    ("testThatItCanResolveWithAutoWiring", testThatItCanResolveWithAutoWiring),
    ("testThatItCanNotResolveWithAutoWiring", testThatItCanNotResolveWithAutoWiring)
  ]
}

// swiftlint:disable trailing_comma
XCTMain([
  testCase(AutoInjectionTests.allTests),
  testCase(AutoWiringTests.allTests),
])
// swiftlint:enable trailing_comma
